tests/returntypes/transforms/test_mitigate_new.py

```diff
@@ -0,0 +1,570 @@
+        assert not np.allclose(res_mitigated, res_ideal)
+
+    def test_single_return(self):
+        """Tests if the expected shape is returned when mitigating a circuit with a single return"""
+        from mitiq.zne.scaling import fold_global
+        from mitiq.zne.inference import RichardsonFactory
+
+        noise_strength = 0.05
+
+        dev_noise_free = qml.device("default.mixed", wires=2)
+        dev = qml.transforms.insert(qml.AmplitudeDamping, noise_strength)(dev_noise_free)
+
+        n_wires = 2
+        n_layers = 2
+
+        shapes = qml.SimplifiedTwoDesign.shape(n_wires, n_layers)
+        np.random.seed(0)
+        w1, w2 = [np.random.random(s) for s in shapes]
+
+        @qml.transforms.mitigate_with_zne([1, 2, 3], fold_global, RichardsonFactory.extrapolate)
+        @qml.qnode(dev)
+        def mitigated_circuit(w1, w2):
+            qml.SimplifiedTwoDesign(w1, w2, wires=range(2))
+            return qml.expval(qml.PauliZ(0))
+
+        @qml.qnode(dev_noise_free)
+        def ideal_circuit(w1, w2):
+            qml.SimplifiedTwoDesign(w1, w2, wires=range(2))
+            return qml.expval(qml.PauliZ(0))
+
+        res_mitigated = mitigated_circuit(w1, w2)
+        res_ideal = ideal_circuit(w1, w2)
+
+        assert res_mitigated.shape == res_ideal.shape
+        assert not np.allclose(res_mitigated, res_ideal)
+
+    def test_with_reps_per_factor(self):
+        """Tests if the expected shape is returned when mitigating a circuit with a reps_per_factor
+        set not equal to 1"""
+        from mitiq.zne.scaling import fold_gates_at_random
+        from mitiq.zne.inference import RichardsonFactory
+
+        noise_strength = 0.05
+
+        dev_noise_free = qml.device("default.mixed", wires=2)
+        dev = qml.transforms.insert(qml.AmplitudeDamping, noise_strength)(dev_noise_free)
+
+        n_wires = 2
+        n_layers = 2
+
+        shapes = qml.SimplifiedTwoDesign.shape(n_wires, n_layers)
+        np.random.seed(0)
+        w1, w2 = [np.random.random(s) for s in shapes]
+
+        @qml.transforms.mitigate_with_zne(
+            [1, 2, 3], fold_gates_at_random, RichardsonFactory.extrapolate, reps_per_factor=2
+        )
+        @qml.qnode(dev)
+        def mitigated_circuit(w1, w2):
+            qml.SimplifiedTwoDesign(w1, w2, wires=range(2))
+            return qml.expval(qml.PauliZ(0))
+
+        @qml.qnode(dev_noise_free)
+        def ideal_circuit(w1, w2):
+            qml.SimplifiedTwoDesign(w1, w2, wires=range(2))
+            return qml.expval(qml.PauliZ(0))
+
+        res_mitigated = mitigated_circuit(w1, w2)
+        res_ideal = ideal_circuit(w1, w2)
+
+        assert res_mitigated.shape == res_ideal.shape
+        assert not np.allclose(res_mitigated, res_ideal)
+
+    def test_integration(self):
+        """Test if the error of the mitigated result is less than the error of the unmitigated
+        result for a circuit with known expectation values"""
```

Comment:
Updated this test to stack the results of `exact_qnode` and `noisy_qnode`.

Reply:
Similarly: why?

Reply:
Same as above: `circuit` is an untransformed QNode with multiple measurements.